src/frontend/src/common/styles/iconStyle.ts

```diff
@@ -1,33 +1,117 @@
 // includes shared styles for icons
 import styled from "@emotion/styled";
-import { getColors, getIconSizes, getSpaces } from "czifui";
-import EditIcon from "src/common/icons/IconEditSmall.svg";
-import TrashIcon from "src/common/icons/IconTrashCanSmall.svg";
+import { CommonThemeProps, getColors, getSpaces, IconButton } from "czifui";
 
-export const StyledTrashIcon = styled(TrashIcon)`
+const iconFillError = (props: CommonThemeProps) => {
+  const colors = getColors(props);
+  return `
+    svg {
+      fill: ${colors?.error[400]};
+    }
+  `;
+};
+
+const iconFillGray = (props: CommonThemeProps) => {
+  const colors = getColors(props);
+  return `
+    svg {
+      fill: ${colors?.gray[500]};
+    }
+  `;
+};
+
+const rightMarginM = (props: CommonThemeProps) => {
+  const spaces = getSpaces(props);
+  return `
+    margin-right: ${spaces?.m}px;
+  `;
+};
+
+// Icon Wrappers change icon colors. Additional attributes can be added in other styled components.
+export const ErrorIconWrapper = styled.div`
+  ${iconFillError}
+`;
+
+export const IconWrapperGray500 = styled.div`
+  ${iconFillGray}
+`;
+
+export const InfoIconWrapper = styled.div`
+  ${iconFillGray}
+  ${(props) => {
+    const colors = getColors(props);
+    return `
+      :hover {
+        svg {
+          fill: ${colors?.primary[400]};
+        }
+      }
+    `;
+  }}
+`;
+
+export const NavIconWrapper = styled.div`
+  svg {
+    fill: white;
+  }
+`;
+
+export const WarningIconWrapper = styled.div`
   ${(props) => {
     const colors = getColors(props);
-    const iconSizes = getIconSizes(props);
+
+    return `
+      svg {
+        fill: ${colors?.warning[600]};
+      }
+    `;
+  }}
+`;
+
+// Styled Icon Wrappers include attributes other than color, i.e. spacing
+export const StyledEditIconWrapper = styled.div`
+  ${rightMarginM}
+
+  svg {
+    fill: black;
+  }
+`;
+
+export const StyledIconWrapperWhite = styled.div`
+  display: flex;
+  svg {
+    fill: white;
+  }
+`;
+
+export const StyledLockIconWrapper = styled.div`
+  ${(props) => {
     const spaces = getSpaces(props);
 
     return `
-      fill: ${colors?.error[400]};
-      height: ${iconSizes?.xs.height}px;
-      width: ${iconSizes?.xs.width}px;
-      margin-right: ${spaces?.m}px;
+      margin-right: ${spaces?.xxs}px;
     `;
   }}
 `;
 
-export const StyledEditIcon = styled(EditIcon)`
-  fill: black;
+export const StyledTrashIconWrapper = styled.div`
+  ${rightMarginM}
+  ${iconFillError}
+`;
+
+export const StyledTreeIconWrapper = styled.div`
```

Comment:
It looks like this one is only used in one place. Is there a benefit to putting it here rather than in the component specific style file?

Reply:
I liked the idea of keeping icon styles together since they use some of the same mixins.  I don't feel strongly either way.  It is a bit more modular to have icons styled in their respective components.  It's also more likely to have repetition.  Thoughts?

Reply:
alternatively this file could include the mixins and then icon styling could be in whichever component.  There might be some duplication, but it wouldn't be too much.

Reply:
I like the idea of the mixins living in the shared area and then you can import them into whatever styling files need them!